crashtracker-ffi/tests/collector_windows_test.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2021-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+#![cfg(windows)]
+
+use anyhow::Context;
+use std::path::PathBuf;
+use std::{env, fs, process};
+
+#[test]
+fn test_crash() {
+    let profile = if cfg!(debug_assertions) {
+        "debug"
+    } else {
+        "release"
+    };
+
+    println!("Profile: {:?}", profile);
+
+    let tmpdir = tempfile::TempDir::new().unwrap();
+    let dirpath = tmpdir.path();
+    let crash_path = dirpath.join("crash");
+
+    let test_app_path = get_artifact_dir().join(profile).join("test_app.exe");
+
+    println!("Test app path: {:?}", test_app_path);
+
+    let output = process::Command::new(test_app_path)
+        .arg(crash_path.to_str().unwrap())
+        .output()
+        .unwrap();
+
+    println!("Test output: {}", String::from_utf8_lossy(&output.stdout));
+
+    assert!(!output.status.success());
+
+    let crash_report = fs::read(crash_path)
+        .context("reading crash report")
+        .unwrap();
+    let crash_payload = serde_json::from_slice::<serde_json::Value>(&crash_report)
+        .context("deserializing crash report to json")
+        .unwrap();
+
+    println!("Crash payload: {:?}", crash_payload);
+
+    assert_eq!(&crash_payload["error"]["is_crash"], true);
+    assert_eq!(&crash_payload["error"]["kind"], "Panic");
+    assert_eq!(&crash_payload["incomplete"], false);
+    assert_eq!(&crash_payload["metadata"]["library_name"], "test_library");
+    assert_eq!(
+        &crash_payload["metadata"]["library_version"],
+        "test_version"
+    );
+    assert_eq!(&crash_payload["metadata"]["family"], "test_family");
+}
+
+fn get_artifact_dir() -> PathBuf {
+    // This variable contains the path in which cargo puts it's build artifacts
+    // This relies on the assumption that the current binary is assumed to not have been moved from
+    // its directory
+    // If the CARGO_TARGET_DIR env var is set, then just use that.
+    if let Ok(env_target_dir) = env::var("CARGO_TARGET_DIR") {
+        return PathBuf::from(env_target_dir);
+    }
+
+    let test_bin_location = PathBuf::from(env::args().next().unwrap());
+    let mut location_components = test_bin_location.components().rev().peekable();
+    loop {
+        let Some(c) = location_components.peek() else {
+            break;
+        };
```

Comment:
`while let Some(c) = location_components.peek() ...`

Reply:
Also, why is peek() needed at all?  Why can't this be 
```rust
for c in test_bin_location.components().rev()
...
```

Edit: I think I see, you want to leave the "target" string still there

Reply:
This is copy/pasted from bin_tests: https://github.com/DataDog/libdatadog/blob/9a4a79118bcf5e62f90eb93b0540ffbf9279ba98/bin_tests/src/lib.rs#L77